src/integrationTest/java/integration/tests/StatementCancelTest.java

```diff
@@ -0,0 +1,106 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+import com.firebolt.jdbc.exception.FireboltException;
+import com.firebolt.jdbc.statement.FireboltStatement;
+
+import integration.IntegrationTest;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+class StatementCancelTest extends IntegrationTest {
+
+	@BeforeEach
+	void beforeEach() {
+		executeStatementFromFile("/statements/statement-cancel/ddl.sql");
+	}
+
+	@AfterEach
+	void afterEach() {
+		executeStatementFromFile("/statements/statement-cancel/cleanup.sql");
+	}
+
+	@Test
+	@Timeout(value = 2, unit = TimeUnit.MINUTES)
+	void shouldCancelQuery() throws SQLException, InterruptedException {
+		String tableName = extractTableName();
+		long totalRecordsToInsert = 1000000000L;
+		String query = String.format("INSERT INTO %s SELECT id FROM generateRandom('id Int8') LIMIT %d", tableName,
+				totalRecordsToInsert);
+		try (Connection connection = createConnection()) {
+			try (FireboltStatement insertStatement = (FireboltStatement) connection.createStatement()) {
+				this.setParam(connection, "use_standard_sql", "0");
+				Thread thread = new Thread(() -> {
+					try {
+						insertStatement.execute(query);
+					} catch (FireboltException e) {
+						if (!e.getType().equals(REQUEST_FAILED)) {
+							throw new RuntimeException(e);
+							// REQUEST_FAILED is expected since the query was aborted
+						}
+					} catch (SQLException e) {
+						throw new RuntimeException(e);
+					}
+				});
+				thread.start();
+				while (!insertStatement.isStatementRunning()) {
+					Thread.sleep(1000);
+				}
+				insertStatement.cancel();
+			}
+			Thread.sleep(5000);
+			verifyThatNoMoreRecordsAreAdded(connection, tableName, totalRecordsToInsert);
+		}
+	}
+
+	private void verifyThatNoMoreRecordsAreAdded(Connection connection, String tableName, long totalRecordsToInsert)
+			throws SQLException, InterruptedException {
+		String countAddedRecordsQuery = String.format("SELECT COUNT(*) FROM %s", tableName);
+		try (Statement countStatement = connection.createStatement()) {
+			ResultSet rs = countStatement.executeQuery(countAddedRecordsQuery);
+			rs.next();
+			long count = rs.getInt(1);
+			log.info("{} records were added to table {} before the statement got cancelled", count, tableName);
+			Thread.sleep(5000); // waiting to see if more records are being added
+			rs = countStatement.executeQuery(countAddedRecordsQuery);
+			rs.next();
+			assertEquals(count, rs.getInt(1));
+			assertTrue(count < totalRecordsToInsert, "No new records were added following the cancellation");
+			rs.close();
+		}
+
+	}
+
+	private String extractTableName() throws SQLException {
+		String tableName = null;
+		try (Connection connection = this.createConnection(); Statement statement = connection.createStatement()) {
+			this.setParam(connection, "use_standard_sql", "0");
+			ResultSet resultSet = statement.executeQuery("SHOW tables");
```

Comment:
I think `SHOW tables` works without use_standard_sql override.

Reply:
Same in some other places.

Reply:
Ok fair, let's pop a comment there so we remember that's how it works.

Reply:
ok 👍 I changed the class a bit so that it makes more sense (changed method name and I'm reusing the connection)